src/lib/chawathe98_matcher.rs

```diff
@@ -46,35 +46,2262 @@
+    );
+    n4.make_child_of(n2, &mut arena).unwrap();
+
+    let n1 = arena.new_node(
+        String::from("INT"),
+        String::from("1"),
+        None,
+        None,
+        None,
+        None,
+    );
+    n1.make_child_of(root, &mut arena).unwrap();
+    arena
+}
+
+/// Creates a simple tree for tests which is ToNodeId
+pub fn create_arena_2() -> Arena<String, ToNodeId> {
+    let mut arena = Arena::new();
+    let root = arena.new_node(
+        String::from("Expr"),
+        String::from("+"),
+        None,
+        None,
+        None,
+        None,
+    );
+
+    let n2 = arena.new_node(
+        String::from("Expr"),
+        String::from("*"),
+        None,
+        None,
+        None,
+        None,
+    );
+    n2.make_child_of(root, &mut arena).unwrap();
+    let n3 = arena.new_node(
+        String::from("INT"),
+        String::from("3"),
+        None,
+        None,
+        None,
+        None,
+    );
+    n3.make_child_of(n2, &mut arena).unwrap();
+    let n4 = arena.new_node(
+        String::from("INT"),
+        String::from("4"),
+        None,
+        None,
+        None,
+        None,
+    );
+    n4.make_child_of(n2, &mut arena).unwrap();
+
+    let n1 = arena.new_node(
+        String::from("INT"),
+        String::from("1"),
+        None,
+        None,
+        None,
+        None,
+    );
+    n1.make_child_of(root, &mut arena).unwrap();
+
+    arena
+}
+
+/// Test that two nodes have the same label and type.
+fn eq<T: Clone + Debug + Eq>(
+    n1: &NodeId<FromNodeId>,
+    arena1: &Arena<T, FromNodeId>,
+    n2: &NodeId<ToNodeId>,
+    arena2: &Arena<T, ToNodeId>,
+) -> bool {
+    arena1[*n1].label == arena2[*n2].label && arena1[*n1].ty == arena2[*n2].ty
+}
+
+/// Test that two nodes have the same label.
+fn eq_label<T: Clone + Debug + Eq>(
+    n1: &NodeId<FromNodeId>,
+    arena1: &Arena<T, FromNodeId>,
+    n2: &NodeId<ToNodeId>,
+    arena2: &Arena<T, ToNodeId>,
+) -> bool {
+    arena1[*n1].ty == arena2[*n2].ty
+}
+
+/// Match two trees and return a store of mappings between them.
+///
+/// This trait should usually be implemented on configuration objects that
+/// define thresholds and weights for a given algorithm.
+pub trait MatchTrees2<T: Clone + Debug> {
+    /// Match two trees and return a store of mappings between them.
+    fn match_trees(&self, base: Arena<T, FromNodeId>, diff: Arena<T, ToNodeId>)
+                   -> MappingStore2<T>;
+
+    /// Describe the matcher for the user.
+    ///
+    /// This is the string that is printed when the user passes in the --list
+    /// CLI option.
+    fn describe(&self) -> String;
+
+    /// This shows the trait for the pruning for the induced edges.
+    /// The result should logically reduce the number of edges produced in induced edges.
+    fn prune_edges(&self, input: MappingStore2<T>) -> (Vec<Edge>, Vec<Edge>);
+}
+
+#[cfg(test)]
+
+mod tests {
+    use super::*;
+
+    #[test]
+    /// Test for perfect Matching
+    /// TEST 1
```

Comment:
Let's remove `TEST 1` at least; and put the other line of comment above the `#[test]`.